Jclouds 1615

ChJung1 opened the pull request:
Hi @gaul,
This is my first github pull request, so forgive me if i did something the wrong way.
I tried to add a test, but feel that it kind of doesn't test the core of the problem. 
Feedback welcome.
    ChJung1

gaul commented on providers/azureblob/src/test/java/org/jclouds/azureblob/blobstore/AzureBlobStoreTest.java:
```diff
@@ -0,0 +1,57 @@
+       String blockId = AzureBlobStore.makeBlockId(partNumber);
+       // if (!blockId.matches(VALIDATION_REGEX)){
+       //    System.out.println("--------------------------------------------- " + partNumber + " -> '" + blockId + "' -> " + blockId.matches(VALIDATION_REGEX));
+       //}
```
Please remove this commented code.

ChJung1 replied:
... removed

gaul commented on providers/azureblob/src/test/java/org/jclouds/azureblob/blobstore/AzureBlobStoreTest.java:
```diff
@@ -0,0 +1,57 @@
+@Test(groups = "unit", testName = "AzureBlobStore")
+public class AzureBlobStoreTest {
+    
+    static String VALIDATION_REGEX = "[a-zA-Z0-9\\-_=]*";
```
Can you make this `private` and `final` and use `Pattern.compile?`?

ChJung1 replied:
... i did so ... hope you like it.

gaul reviewed (changes requested):
Please squash commits.

gaul commented:
Thank you for your contribution @ChJung1!

inno-steffg commented:
I have tested some simple things with the current JCloud master branch code base against an Azure storage account and run into an error message for a 4 gigabyte upload, split into 5 MByte chunks in a multipart upload:
![2023-07-25_11h40_04](https://github.com/apache/jclouds/assets/5442481/67a1fc92-5691-485e-9624-b3e8627966d4)

Could it be, that the base64 encoding in the url version isn't accepted by Azure?

ChJung1 commented:
Hi Steffen Gürtler,
I did the pull request for the encoding with Base64Url, because with Base64 i got an error similar to yours.

Base64Url is meant to ensure you can use the string in urls.

The string you have contains a minus sign, so this is indeed Base64Url, and not Base64, so my fix is in there.

Your problem looks very similar to mine, so is it maybe the "==" this time?

 

BTW, while changing the code i wondered if we could implement makeBlockId() by using String.valueOf() instead,

which is much simpler code, a shorter string, and uses only digits (as Block-ID is an integer and probably positive)

Feel free to try that, and if it helps you can maybe also remove that not-so-helpful test I added.

 

Andrew Gaul will then hopefully approve your pull request.

 

Good luck!

    ChJung1

 

Gesendet: Dienstag, 25. Juli 2023 um 18:07 Uhr
Von: "Steffen Gürtler" ***@***.***>
An: "apache/jclouds" ***@***.***>
Cc: "ChJung1" ***@***.***>, "Mention" ***@***.***>
Betreff: Re: [apache/jclouds] Jclouds 1615 (PR #155)


 

I have tested some simple things with the current JCloud master branch code base against an Azure storage account and run into an error message for a 4 gigabyte upload, split into 5 MByte chunks in a multipart upload:


Could it be, that the base64 encoding in the url version isn't accepted by Azure?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

gaul commented:
Please open a Jira issue for this regression.
